GPUSolver/CMakeLists.txt

```diff
@@ -0,0 +1,34 @@
+cmake_minimum_required(VERSION 3.22)
+project(GPUSolver)
+
+set(CMAKE_CXX_STANDARD 20)
```

Comment:
Just note that C++20 is only supported on certain devices like A40. For example, NVIDIA Orin only supports C++ ≤17. It's best to write your C++ code with this in mind.